components/style/invalidation/element/invalidator.rs

```diff
@@ -500,9 +503,9 @@ where
                 self.invalidate_pseudo_element_or_nac(before, invalidations);
         }
 
-        let node = self.element.as_node();
+        let element = self.element;
         any_descendant |=
-            self.invalidate_dom_descendants_of(node, invalidations);
+            self.invalidate_dom_descendants_of(element, invalidations);
```

Comment:
Use `self.element` here and remove the `let element` above.

Reply:
I can't, due to the borrow checker and the lack of nll.

Reply:
Hmm, maybe I can, but I think I still want to iterate over the DOM children of a doc fragment for querySelector, so I'll refactor the patch a bit.